spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/neo4j/Neo4jDriverPropertiesTests.java

```diff
@@ -0,0 +1,416 @@
+/*
+ * Copyright 2012-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.neo4j;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.boot.autoconfigure.neo4j.Neo4jDriverProperties.Authentication;
+import org.springframework.boot.autoconfigure.neo4j.Neo4jDriverProperties.DriverSettings;
+import org.springframework.boot.autoconfigure.neo4j.Neo4jDriverProperties.PoolSettings;
+import org.springframework.boot.autoconfigure.neo4j.Neo4jDriverProperties.TrustSettings;
+import org.springframework.boot.autoconfigure.neo4j.Neo4jDriverProperties.TrustSettings.Strategy;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.context.properties.source.InvalidConfigurationPropertyValueException;
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Configuration;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.time.Duration;
+
+import org.neo4j.driver.AuthTokens;
+import org.neo4j.driver.Config;
+import org.neo4j.driver.internal.retry.RetrySettings;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+/**
+ * @author Michael J. Simons
+ */
+class Neo4jDriverPropertiesTests {
+
+	private AnnotationConfigApplicationContext context;
+
+	private Neo4jDriverProperties load(String... properties) {
+
+		AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext();
```

Comment:
Even such tests use the application context runner. It is a superior option compared to a simple handling of the context as you still need to handle the context otherwise.

Given that the logic for the properties object would be moved elsewhere, this might not be necessary after all.